gc/src/lib.rs

```diff
@@ -60,19 +60,43 @@ impl<T: Trace> Gc<T> {
     /// let five = Gc::new(5);
     /// ```
     pub fn new(value: T) -> Self {
+        debug_assert!(mem::align_of::<GcBox<T>>() > 1);
```

Comment:
Let's make this an assert!(). It will be optimized out of release builds anyway.